libr/core/canal.c

```diff
@@ -2519,12 +2519,39 @@ static int fcn_print_detail(RCore *core, RAnalFunction *fcn) {
 	return 0;
 }
 
+static bool is_fcn_traced(RDebugTrace *traced, RAnalFunction *fcn) {
+	int tag = traced->tag;
+	RListIter *iter;
+	RDebugTracepoint *trace;
+
+	r_list_foreach (traced->traces, iter, trace) {
```

Comment:
just use this api `R_API RDebugTracepoint *r_debug_trace_get(RDebug *dbg, ut64 addr); ` to check if the first instruction has been traced or not. this should be enough to determine if the function ist raced.

Reply:
https://github.com/radare/radare2/pull/12798#discussion_r248018901

Reply:
> just use this api `R_API RDebugTracepoint *r_debug_trace_get(RDebug *dbg, ut64 addr); ` to check if the first instruction has been traced or not. this should be enough to determine if the function ist raced.

`e dbg.trace=1`
`db 0x4004d7`
`dc` -> hits breakpoint
`dso`

in this above case checking only the first instruction won't work. we need to go through the debug trace list of addresses and check if it exists in the fcn, if yes then fcn is being traced.